### ISSUE TITLE
save computation by just `.detach()`ing the previous outputs

### ISSUE THREAD

**Issue by Jemoka** [alliegriffith/perez_reimplementation#14]
i.e., you can implement stopgradient but won't hurt the thing you called detached on; i.e.

```python
i_hate_gradients = i_love_gradients.detach()
do_things_with += i_hate_gradients

i_love_gradients.backward()
```

https://github.com/alliegriffith/perez_reimplementation/blob/e36436c3bc2e6664b79564a31dc5ecf2f55ae3c7/perez.py#L281-L292
